### DISCUSSION TITLE
ogmios: DecoderFailure

### DISCUSSION THREAD

**Discussion by NanuIjaz** [CardanoSolutions/ogmios#371] [Q & A] [answered]
there is an error in ogmios ,

ogmios: DecoderFailure (ChainSync (HardForkBlock (': * ByronBlock (': * (ShelleyBlock (TPraos StandardCrypto) (ShelleyEra StandardCrypto)) (': * (ShelleyBlock (TPraos StandardCrypto) (AllegraEra StandardCrypto)) (': * (ShelleyBlock (TPraos StandardCrypto) (MaryEra StandardCrypto)) (': * (ShelleyBlock (TPraos StandardCrypto) (AlonzoEra StandardCrypto)) (': * (ShelleyBlock (Praos StandardCrypto) (BabbageEra StandardCrypto)) (': * (ShelleyBlock (Praos StandardCrypto) (ConwayEra StandardCrypto)) ('[] *))))))))) (Tip HardForkBlock (': * ByronBlock (': * (ShelleyBlock (TPraos StandardCrypto) (ShelleyEra StandardCrypto)) (': * (ShelleyBlock (TPraos StandardCrypto) (AllegraEra StandardCrypto)) (': * (ShelleyBlock (TPraos StandardCrypto) (MaryEra StandardCrypto)) (': * (ShelleyBlock (TPraos StandardCrypto) (AlonzoEra StandardCrypto)) (': * (ShelleyBlock (Praos StandardCrypto) (BabbageEra StandardCrypto)) (': * (ShelleyBlock (Praos StandardCrypto) (ConwayEra StandardCrypto)) ('[] *)))))))))) ServerAgency TokNext TokMustReply) (DeserialiseFailure 1771 "expected list len or indef")

ogmios version is - cardanosolutions/ogmios:v6.0.0-rc6
this is for sanchonet 
node version is cardano-node:8.8.0-pre

what could be the reason?

**Comment by KtorZ** [comment, marked as answer]
rc6 isn't compatible with current SanchoNet. The Conway era in the node is still a work in progress and the serialisation codecs are changing in a non-backward compatible way on every node release. 

Ogmios is doing its best to keep up with the release cadence, but old versions aren't guaranteed to work with the latest era still in development. 

So TL;DR; always use the latest version and check the README to know the compatibility matrix between Ogmios and the node.
